pkg/bootstrap/platform/gcp.go

```diff
@@ -365,9 +373,15 @@ type GcpInstance struct {
 
 // Checks metadata to see if GKE metadata or Kubernetes env vars exist
 func (e *gcpEnv) IsKubernetes() bool {
-	md := e.Metadata()
+	// sometimes for non-gcp platforms, metadataIP doesn't return anything. It's
+	// safer to use the existing env variable.
+	_, _, cluster, _ := parseGCPMetadata()
+	if cluster == "" {
+		md := e.Metadata()
+		cluster = md[GCPCluster]
+	}
 	_, onKubernetes := os.LookupEnv(KubernetesServiceHost)
```

Comment:
nit: can we check this first and return early if its true? Then we can avoid Metadata() lookup. can be followup